### PR TITLE
Create `PromptExecutionHistory` entity

### LINKED ISSUE #79: Create `PromptExecutionHistory` entity
### Description
An entity that stores the AnalysisPrompt's execution history. 

### Requirements
- project should be runnable
- must inherit from `Entity` type

### Deliverables
- `PromptExecutionHistory` entity, dto, mapper, repository, configuration and migration, validator, service and controller endpoints with only create and get methods.

### PR DESCRIPTION
**Pull request by bunyodkamalov**
Closes #79

### CODE REVIEW

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/PromptsHistory/Models/PromptExecutionHistoryDto.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    public Guid PromptId { get; set; }
+
+    public string? Result { get; set; }
+
+    public string? Exception { get; set; } 
+    
+    public DateTime ExecutionTime { get; set; } 
+    
+    public int ExecutionDurationInSeconds { get; set; }
+}
```
Comment (botirjonabdumannonovv) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/PromptsHistory/Models/PromptExecutionHistoryDto.cs:16]: write docs

**Review by botirjonabdumannonovv — DISMISSED** [review]

**Review by rashidovasadbek — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/PromptExecutionHistory.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    /// <summary>
+    /// Gets or sets the exception encountered during prompt execution, if any.
+    /// </summary>
+    public string? Exception { get; set; } 
+    
+    /// <summary>
+    /// Gets or sets the time when the prompt was executed.
+    /// </summary>
+    public DateTime ExecutionTime { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/PromptExecutionHistory.cs:28]: Change to `DateTimeOffset`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/PromptsHistory/Models/PromptExecutionHistoryDto.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    
+    public Guid PromptId { get; set; }
+
+    public string? Result { get; set; }
+
+    public string? Exception { get; set; } 
+    
+    public DateTime ExecutionTime { get; set; } 
+    
+    public int ExecutionDurationInSeconds { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/PromptsHistory/Models/PromptExecutionHistoryDto.cs:15]: Change to `ExecutionDurationInMilliseconds`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/PromptsHistory/Queries/PromptExecutionHistoryGetByPromptIdQuery.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+using Feedback.Analyzer.Domain.Common.Queries;
+
+namespace Feedback.Analyzer.Application.PromptsHistory.Queries;
+
+/// <summary>
+/// Represents a query for retrieving prompt execution history records by prompt ID.
+/// </summary>
+public record PromptExecutionHistoryGetByPromptIdQuery : IQuery<IList<PromptExecutionHistoryDto>>
+{
+    public Guid PromptId { get; set; } = default!;
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/PromptsHistory/Queries/PromptExecutionHistoryGetByPromptIdQuery.cs:11]: Add docs

**Review by sulton-max — CHANGES_REQUESTED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/PromptExecutionHistory.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    
+    /// <summary>
+    /// Gets or sets the time when the prompt was executed.
+    /// </summary>
+    public DateTimeOffset ExecutionTime { get; set; } 
+    
+    /// <summary>
+    /// Gets or sets the duration of the prompt execution.
+    /// </summary>
+    public TimeSpan ExecutionDurationInMilliSeconds { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/PromptExecutionHistory.cs:33]: Change to `ExecutionDuration`

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — APPROVED** [review]
Good job 👏

**Review by isxoqovxasanboy — APPROVED** [review]

**Pull request merged**